users/views.py

```diff
@@ -1,3 +1,34 @@
-from django.shortcuts import render
+import requests, jwt
 
-# Create your views here.
+from django.views import View
+from django.http  import JsonResponse
+from django.conf  import settings
+from datetime     import datetime, timedelta
+
+from users.models import User
+
+
+class KakaoSignInView(View):
+    def get(self, request):
+        try:
+            kakao_token = request.headers.get('Authorization')
+            headers     = {'Authorization' : f'Bearer {kakao_token}'}
+            response    = requests.get('https://kapi.kakao.com/v2/user/me', headers=headers, timeout=3)
+            kakao_user  = response.json()
+            kakao_id    = kakao_user['id']
+            email       = kakao_user['kakao_account']['email']
+            name        = kakao_user['kakao_account']['profile']['nickname']
+            
+            user, created = User.objects.get_or_create(
+                kakao_id  = kakao_id,
+                email     = email,
+                name      = name
+            )
```

Comment:
- get_or_create의 2번째 return 값은 boolean이기 때문에 변수 앞에 is_을 붙여주는 것이 좋습니다.
- get_or_create을 사용할 때는 pk 값으로 사용 되는 값으로만 객체를 특정 지어주어야 합니다. 그렇지 않으면 조건을 모두 만족하는 객체를 찾기 때문에 값이 조그만 달라도 새로운 객체를 만들어 버립니다.

```suggestion
            user, is_created = User.objects.get_or_create(
                kakao_id  = kakao_id,
            )
```

Reply:
넵 리뷰 반영해서 코드 수정하겠습니다!!